src/deprecatedCompat/4.8/mergeDecoratorsAndModifiers.ts

```diff
@@ -496,7 +497,7 @@ function patchNodeFactory(factory: NodeFactory) {
                 (name === undefined || typeof name === "string" || isPropertyName(name)) &&
                 (questionToken === undefined || typeof questionToken === "object" && isQuestionToken(questionToken)) &&
                 (typeParameters === undefined || isArray(typeParameters)) &&
-                (parameters === undefined || !some(parameters, isTypeParameterDeclaration)) &&
+                (parameters === undefined || !some<Node>(parameters, isTypeParameterDeclaration)) &&

```

Comment:
These are all cases that work really really oddly. They arguably shouldn't need this manually, but, it gets really really screwy:

![infernodewhat](https://user-images.githubusercontent.com/5341706/211222256-038c55ae-b3be-45ca-ac0b-af94bb2e7dac.gif)

Reply:
Fixed in main; I've reverted all of these with the expectation that we will bump LKG to a newer version of 5.0.